### PR TITLE
JSONize gunmod weight multiplier

### PR DESCRIPTION
**Pull request by pawnishoovy**
#### Summary

```SUMMARY: Infrastructure "JSONize gunmod weight multiplier'"```

#### Purpose of change

Gunmods such as lightweight replacement furniture are hardcoded at a weight multiplier of 0.75 and this cannot be changed through JSON alone.

#### Describe the solution

This PR removes the REDUCED_WEIGHT flag from lightweight replacement furniture and instead adds a new JSON variable "weight_multiplier" that affects gun weight in the same way REDUCED_WEIGHT did.

REDUCED_WEIGHT has NOT been removed from the game and may still be in use elsewhere.

#### Describe alternatives you've considered

N/A

#### Testing

Take a gun and lightweight replacement furniture
Apply the furniture gunmod
See as gun weight is affected despite no REDUCED_WEIGHT flag

#### Additional context

This will allow even more innate-weight-affecting gunmods with different values. For a hypothetical, a steel-to-titanium gunmod that affects multiplies weight by 0.5 instead of 0.75.

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
Did you test if the weight of _other_ attached gunmods gets altered by this mod?

**Comment by pawnishoovy** [comment]
> Did you test if the weight of _other_ attached gunmods gets altered by this weight multiplier?

It does not change the weight of other mods, i.e. it works correctly. Just tested now as we speak.

**Comment by pawnishoovy** [comment]
Unsure what's wrong with the travis check, I've astyled everything with 3.0.1 and the options given in the instructions.

File: src/itype.h [hunk 1]
```diff
@@ -590,6 +590,9 @@ struct islot_gunmod : common_ranged_data {
 
     /** Increases base gun UPS consumption by this many times per shot */
     float ups_charges_multiplier = 1.0f;
+
```
Comment (anothersimulacrum) [inline src/itype.h]:
```suggestion

```

**Pull request merged**